### PR TITLE
Add support for plain text documents

### PR DESCRIPTION
**Pull request by evanxg852000**
Closes https://github.com/quickwit-oss/quickwit/issues/3017

### CODE REVIEW

File: quickwit/quickwit-config/src/source_config/mod.rs [hunk 1]
```diff
@@ -79,6 +79,9 @@ pub struct SourceConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "transform")]
     pub transform_config: Option<TransformConfig>,
+
+    // Denotes if the data is in json or any other format.
+    pub is_json_format: bool,
```
Comment (guilload) [inline quickwit/quickwit-config/src/source_config/mod.rs:84]: Let's use an `data_format` (or `input_format`?) enum for this parameter that defaults to JSON:
```rust
#[derive(Debug, Clone, Copy, Default, ...)]
enum SourceDataFormat {
  #[default]
  Json,
  #[serde(alias = "plain")]
  PlainText, 
}
```

**Comment by fulmicoton** [comment]
@fmassot @guilload  @evanxg852000  what is the status on this?

**Comment by guilload** [comment]
I'll take this PR over and get it merged.

File: quickwit/quickwit-indexing/src/actors/doc_processor.rs [hunk 2]
```diff
@@ -321,22 +388,22 @@ impl Handler<RawDocBatch> for DocProcessor {
             return Ok(());
         }
         let mut prepared_docs: Vec<PreparedDoc> = Vec::with_capacity(raw_doc_batch.docs.len());
-        for json_doc in raw_doc_batch.docs {
-            let json_doc_num_bytes = json_doc.len() as u64;
+        for doc in raw_doc_batch.docs {
+            let doc_num_bytes = doc.len() as u64;
 
-            match self.prepare_document(&json_doc, ctx) {
+            match self.prepare_document(doc, ctx) {
```
Comment (fmassot) [inline quickwit/quickwit-indexing/src/actors/doc_processor.rs:394]: nitpick: could be `process_document` that returns a `ProcessedDoc` to be consistend with `DocProccessorError`.

**Review by fmassot — APPROVED** [review]
LGTM

File: quickwit/quickwit-indexing/src/actors/doc_processor.rs (hunk 2 above)
Comment (guilload) [reply quickwit/quickwit-indexing/src/actors/doc_processor.rs:394]: Indeed. Will fix.

**Pull request merged**